airbyte-integrations/connectors/source-salesforce/unit_tests/integration/test_bulk_stream.py

```diff
@@ -65,29 +72,225 @@ def tearDown(self) -> None:
+            HttpResponse(json.dumps({"id": _JOB_ID})),
+        )
+        self._http_mocker.get(
+            HttpRequest(f"{_BASE_URL}/jobs/query/{_JOB_ID}"),
+            SalesforceJobResponseBuilder().with_id(_JOB_ID).with_state("Failed").build(),
+        )
+        self._http_mocker.get(
+            create_standard_http_request(_STREAM_NAME, [_A_FIELD_NAME]),
+            create_standard_http_response([_A_FIELD_NAME]),
+        )
+        self._mock_delete_job(_JOB_ID)
+
+        output = read(_STREAM_NAME, SyncMode.full_refresh, self._config)
+
+        assert len(output.records) == 1
+
+    def test_given_retryable_error_on_download_job_result_when_read_then_extract_records(self):
+        given_stream(self._http_mocker, _BASE_URL, _STREAM_NAME, SalesforceDescribeResponseBuilder().field(_A_FIELD_NAME))
+        self._http_mocker.post(
+            self._create_full_job_creation_request([_A_FIELD_NAME]),
+            HttpResponse(json.dumps({"id": _JOB_ID})),
+        )
+        self._http_mocker.get(
+            HttpRequest(f"{_BASE_URL}/jobs/query/{_JOB_ID}"),
+            SalesforceJobResponseBuilder().with_id(_JOB_ID).with_state("JobComplete").build(),
+        )
+        self._http_mocker.get(
+            HttpRequest(f"{_BASE_URL}/jobs/query/{_JOB_ID}/results"),
+            [
+                _RETRYABLE_RESPONSE,
+                HttpResponse(f"{_A_FIELD_NAME}\nfield_value"),
+            ],
+        )
+        self._mock_delete_job(_JOB_ID)
+
+        output = read(_STREAM_NAME, SyncMode.full_refresh, self._config)
+
+        assert len(output.records) == 1
+
+    def test_given_retryable_error_on_delete_job_result_when_read_then_do_not_break(self):
+        given_stream(self._http_mocker, _BASE_URL, _STREAM_NAME, SalesforceDescribeResponseBuilder().field(_A_FIELD_NAME))
+        self._http_mocker.post(
+            self._create_full_job_creation_request([_A_FIELD_NAME]),
+            HttpResponse(json.dumps({"id": _JOB_ID})),
+        )
+        self._http_mocker.get(
+            HttpRequest(f"{_BASE_URL}/jobs/query/{_JOB_ID}"),
+            SalesforceJobResponseBuilder().with_id(_JOB_ID).with_state("JobComplete").build(),
+        )
+        self._http_mocker.get(
+            HttpRequest(f"{_BASE_URL}/jobs/query/{_JOB_ID}/results"),
+            HttpResponse(f"{_A_FIELD_NAME}\nfield_value"),
+        )
+        self._http_mocker._mock_request_method(  # FIXME to add DELETE method in airbyte_cdk tests
+            "delete",
+            HttpRequest(f"{_BASE_URL}/jobs/query/{_JOB_ID}"),
+            [
+                _RETRYABLE_RESPONSE,
+                HttpResponse(""),
+            ],
+        )
+
+        output = read(_STREAM_NAME, SyncMode.full_refresh, self._config)
+
+        assert output.get_stream_statuses(_STREAM_NAME)[-1] == AirbyteStreamStatus.COMPLETE
+
+    def test_given_non_retryable_error_on_delete_job_result_when_read_then_fail_to_sync(self):
+        """
```

Comment:
I don't think we've seen errors for that in prod but it would be sad kill the stream because of that